src/hotspot/share/opto/doCall.cpp

```diff
@@ -275,9 +295,22 @@ CallGenerator* Compile::call_generator(ciMethod* callee, int vtable_index, bool
           } else {
             // Generate virtual call for class check failure path
             // in case of polymorphic virtual call site.
-            miss_cg = CallGenerator::for_virtual_call(callee, vtable_index);
+            if (PolymorphicInlining && cg_intrinsic != NULL) {
+              miss_cg = cg_intrinsic;
```

Comment:
why use cg_intrinsic for miss_cg?

Reply:
available intrinsic cg is much better than virtual call . Falling back to virtual call has significant regression.